### PR TITLE
providers/common/der/build.info: Improve checks of disabled algos

### PR DESCRIPTION
**Pull request by levitte**
This protects us from unwanted GENERATE statements in particular.

-----

It solves the issue identified in this comment: https://github.com/openssl/openssl/pull/13537#issuecomment-738596108

### CODE REVIEW

**Review by DDvO — APPROVED** [review]
LGTM

File: providers/common/der/build.info [hunk 1]
```diff
@@ -25,43 +25,49 @@ GENERATE[$DER_RSA_H]=der_rsa.h.in
...
+  DEPEND[${DER_EC_GEN/.c/.o}]=$DER_EC_H
+  GENERATE[$DER_EC_H]=der_ec.h.in
+  DEPEND[$DER_EC_H]=oids_to_c.pm
+ENDIF
 
 #----- ECX
-$DER_ECX_H=../include/prov/der_ecx.h
-$DER_ECX_GEN=der_ecx_gen.c
-$DER_ECX_AUX=der_ecx_key.c
+IF[{- !$disabled{ec} -}]
```
Comment (mattcaswell) [inline providers/common/der/build.info:58]: Couldn't we just merge this into the immediately prior "no-ec" block?

Comment (DDvO) [reply providers/common/der/build.info:58]: I also thought about this but the current way is slightly nicer aligned with the structure/comments given in the file.

**Review by afshin2020 — APPROVED** [review]

**Review by afshin2020 — CHANGES_REQUESTED** [review]
Improve

File: providers/common/der/build.info (hunk 1 above)
Comment (levitte) [reply providers/common/der/build.info:58]: What @DDvO said

Comment (mattcaswell) [reply providers/common/der/build.info:58]: ok

**Review by mattcaswell — APPROVED** [review]

**Comment by DDvO** [comment]
@levitte, this is ready to merge, and then #13537 can be rebased on that :)

**Comment by levitte** [comment]
Merged

142841ddc4940b6fe74d8c8a846f27d6e2ee29dd providers/common/der/build.info: Improve checks of disabled algos

**Pull request closed without merging**